addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -316,6 +317,22 @@ def _l10n_tr_nilvera_add_pdf_to_invoice(self, client, invoice, document_uuid, do
         invoice.message_main_attachment_id = attachment
         invoice.with_context(no_new_invoice=True).message_post(attachment_ids=attachment.ids)
 
+    def l10n_tr_nilvera_get_pdf(self):
+        with _get_nilvera_client(self.env.company) as client:
+            invoices_to_get_pdf = self.filtered(
+                lambda invoice: invoice.l10n_tr_nilvera_customer_status in {'einvoice', 'earchive'}
+                                and invoice.message_main_attachment_id.id == invoice.invoice_pdf_report_id.id
+                                and invoice.l10n_tr_nilvera_send_status == 'succeed'
+            )
+            for invoice in invoices_to_get_pdf:
+                self._l10n_tr_nilvera_add_pdf_to_invoice(
+                    client,
+                    invoice,
+                    invoice.l10n_tr_nilvera_uuid,
+                    document_category="Sale",
+                    invoice_channel=invoice.l10n_tr_nilvera_customer_status,
+                )
```

Comment:
Generally better to do a python loop here:

```suggestion
            for invoice in self:
                if invoice.l10n_tr_nilvera_customer_status not in {'einvoice', 'earchive'} or invoice.message_main_attachment_id.id != invoice.invoice_pdf_report_id.id or invoice.l10n_tr_nilvera_send_status != 'succeed':
                    continue
                self._l10n_tr_nilvera_add_pdf_to_invoice(
                    client,
                    invoice,
                    invoice.l10n_tr_nilvera_uuid,
                    document_category="Sale",
                    invoice_channel=invoice.l10n_tr_nilvera_customer_status,
                )
```

Maybe do the condition on several lines?
By doing this we are only doing one loop instead of 2